spdmlib/src/requester/context.rs

```diff
@@ -53,9 +53,20 @@ impl RequesterContext {
             #[cfg(not(feature = "mut-auth"))]
             let req_slot_id: Option<u8> = None;
             #[cfg(feature = "mut-auth")]
-            self.session_based_mutual_authenticate(session_id).await?;
-            #[cfg(feature = "mut-auth")]
-            let req_slot_id = Some(self.common.runtime_info.get_local_used_cert_chain_slot_id());
+            let req_slot_id = {
+                if self
+                    .common
+                    .negotiate_info
+                    .rsp_capabilities_sel
+                    .contains(SpdmResponseCapabilityFlags::MUT_AUTH_CAP)
```

Comment:
I think we need check both rsp_cap and req_cap.

Reply:
> I think we need check both rsp_cap and req_cap.

Yes, Updated.